### ISSUE TITLE
Processing large Belize dataset: S3080_20190802_20200109 and S3080_20190606_20200109

### ISSUE THREAD

**Issue by RossPitman** [pantheracorp/PantheraIDS_Issues#161]
From Emma:

So I’m looking for options on how to proceed with the following. We have a database with S3080_20190802_20200109 in the server with data from Panthera cameras only though this database is part of a larger dataset. The larger grid used both Panthera cameras and Moultrie brand cameras but we were unable to classify the entire dataset with machine classifier, therefore these were spilt. I would like to joint these two so that we are able to analysis the data of the large grid not just a section of it. We currently have a database, named as S3080_20190606_20200109, with the pictures processed containing both Panthera cam and Moultrie data, though I did not send this to be uploaded into the server.
·       I want to know if I would have to ignore this version (S3080_20190606_20200109) and reprocess only the Moultrie data then merge it to S3080_20190802_20200109? Keep in mind that the Moultrie cameras started monitoring before the start date of this survey.
·       Or if I can classify S3080_20190606_20200109 then merge to S3080_20190802_20200109 without having duplicates? The downside of this is that I do not know if my laptop, is what I want to use, would be able to classify this large data (407 GB in images). The Moultrie camera settings allowed for false triggers increasing blanks and memory storage.
·       Can these be processed separately (two surveys from the same sites) even though they overlap in survey dates?
·       What would you recommend is the best way to go?
 
I wanted your opinion, maybe share it with your teammates also, so that I run into having to do double work. In the end, these data need to be analysed. If we can’t do it through IDS, it would be a shame. At the moment I don’t think we would have access to a high power machine to process large data sets like this.  I’ll give it a try but I needed some sort of second opinion and recommendation.

**Comment by RossPitman** [comment]
Further correspondence from Emma:

Hi Shannon,
 
You have a database on the server (S3080_20190802_20200109) which has only Panthera cams, and has been machine classified already. This is a subset of a larger survey. CORRECT

You processed the full survey (S3080_20190606_20200109), which includes the Panthera cams as well as the Moultrie cams, and this is a local database that has not been put through the machine classifier yet. CORRECT

All the information from the Panthera cam subset survey (S3080_20190802_20200109) is included in this larger database (S3080_20190606_20200109). CORRECT

Moving forward the end goal is to have the larger database (S3080_20190606_20200109) completely machine classified and able to be fully analysed within IDS. CORRECT

Secondly….

Are the Stations consistent across these two databases? For example, is Station 1 in the subset database the same station as Station 1 in the full database? Is this true for all stations? Yes, essentially the csv files for the subset data (S3080_20190802_20200109) were extrapolated from the full dataset (S3080_20190606_20200109) cvs files. The S3080_20190606_20200109 was created first and since we were unable to machine classify it in its entirety we proceeded to split.

Are the Cameras consistent across these two databases? For example, is CAM1 in the subset database the same camera as CAM1 in the full database? Or CAM42387 (just throwing out ransom examples)? This should be true for all stations between the subset (S3080_20190802_20200109) and full dataset (S3080_20190606_20200109).

You mentioned that all and only Panthera cams were included in the subset database- 
Assuming that your stations have 2 cameras each, is there any station which has a Panthera cam and a Moultrie cam? If so, is this station included in both databases/how was this station handled?

Is there any station which had a Panthera cam which was then later replaced with a Moultrie cam (or vice cersa)?If so, is this station included in both databases/how was this station handled? No

In response to the above (7), in the subset data (S3080_20190802_20200109) already in IDS, three Moultrie cameras were included in the subset which translates to a double station with Moultrie cams and a single station, not paired at any point with a Panthera camera. The camera numbers are consistent between the subset and full datasets. Field work for the entire grid was surveyed by Panthera Belize and a partner organization. The partner organization used in their “half” of the grid Moultrie cams while we used mostly Panthera cameras (except for those 3 cameras). The portion that is left to be classified is ONLY Moultrie camera data.

As for limitations on machine classifying, there is a way to split the images into batches and machine classify them bit by bit. Then at the end, you would just need to combine the predictions csvs together using simple copy/paste in excel before merging the classifications back into the database through IDS. I will be more than happy to give more specific details when we get to that stage.
-        I checked for the size of files per check on the remaining dataset, the first check is ~60 GB while the second is ~40 GB. I had mentioned that I wanted to use my laptop to run these as this is the only machine I have had luck with processing data. However these two files are way too large for the space I have available on my C-drive. I hope I am able to make another machine work in my favour.
 
Hopefully I have made everything more clear. Database S3080_20190606_20200109 is attached in email.
